### PR TITLE
ODBC: Use literals instead of parameters in Power BI data connector

### PR DESCRIPTION
**Pull request by rupal-bq**
*Issue #651:*

*Description of changes:*
- added support for literals instead of parameters
- disabled direct query
- updated user docs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sql-odbc/src/PowerBIConnector/OdfeSqlOdbcPBIConnector.pq [hunk 1]
```diff
@@ -106,6 +106,20 @@ OdfeSqlOdbcPBIConnectorImpl = (Host as text, optional Port as number) as table =
...
+        SQLGetFunctions = Diagnostics.LogValue("SQLGetFunctions_Options", [
+            SQL_API_SQLBINDPARAMETER = false
+        ]),
+
+        SqlCapabilities = Diagnostics.LogValue("SqlCapabilities_Options", [
+            SupportsTop = false,
+            LimitClauseKind = LimitClauseKind.LimitOffset,
+            SupportsNumericLiterals = true,
+            SupportsStringLiterals = true,
+            SupportsOdbcDateLiterals = true,
```
Comment (jordanw-bq) [inline sql-odbc/src/PowerBIConnector/OdfeSqlOdbcPBIConnector.pq:119]: Can you confirm these options for date/time/timestamp literals? Wondering if this refers to the `{dt ...}`-type syntax we've seen before, which ODFE SQL plugin doesn't support

Comment (rupal-bq) [reply sql-odbc/src/PowerBIConnector/OdfeSqlOdbcPBIConnector.pq:119]: Yes, this refers to `{d ..}` syntax. But if we don't set this to true then date values will always be specified using Parameter Binding (https://docs.microsoft.com/en-us/power-query/odbc) which is also not supported.

I will check if it is possible to report the escape sequence syntax unsupported.

Comment (rupal-bq) [reply sql-odbc/src/PowerBIConnector/OdfeSqlOdbcPBIConnector.pq:119]: ODBC driver sets `SQL_DATETIME_LITERALS` to 0 in SQLGetInfo and a value of "0" means that SQL-92 datetime literals are not supported. (https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetinfo-function?view=sql-server-ver15)

Comment (jordanw-bq) [reply sql-odbc/src/PowerBIConnector/OdfeSqlOdbcPBIConnector.pq:119]: Verified with @rupal-bq that timestamp values work with the current PBI connector; will look back at this when we include DirectQuery support

**Pull request merged**